core/src/main/java/com/linecorp/armeria/server/Route.java

```diff
@@ -92,8 +92,8 @@ static RouteBuilder builder() {
      *
      * {@link RoutePathType#REGEX} has only one path that represents it. e.g, {@code [ "^/(?<foo>.*)$" ]}
      *
-     * <p>{@link RoutePathType#REGEX_WITH_PREFIX} has two paths. The first one is the prefix and the second
-     * one is the regex. e.g, {@code [ "/bar/", "^/(?<foo>.*)$" ]}
+     * <p>{@link RoutePathType#REGEX_WITH_PREFIX} has two paths. The first one is the regex and the second
+     * one is the path. e.g, {@code [ "^/(?<foo>.*)$", "/bar/" ]}
```

Comment:
cc @minwoox
https://github.com/line/armeria/blob/master/core/src/test/java/com/linecorp/armeria/server/PathWithPrefixTest.java#L33

Reply:
Oops, thanks a lot. ❤️

Reply:
can this be combined into one pattern? I don't quite understand why having two helps. the main thing about http.route is the matched route. Is this saying there are two matched routes?

Reply:
Let's say a user specified `regex:x`, which will match every path that contains x. If we concat the prefix and regex without a space, it will be `/foo/x` which is not distinguishable from `/foo/x`.

Reply:
the span name is a grouping concept as well as a user label. I'm not sure if you meant to write /foo/x twice but if you did then the result of the matched route would be the important part. the aim of the http.route is to lower the cardinality of paths.. not really much else.

it may be the case here that flexibility makes these appear more complex than other frameworks but I would sit on the idea a bit and see if there is a way to avail more user friendly route names. this doesnt have to happen in this PR if it is complicated